lib/Runtime/Library/WebAssemblyModule.cpp

```diff
@@ -772,6 +827,32 @@ uint WebAssemblyModule::GetGlobalsByteSize() const
     return size;
 }
 
+void
+WebAssemblyModule::AddCustomSection(Wasm::CustomSection customSection)
+{
+    if (!m_customSections)
+    {
+        m_customSections = Anew(&m_alloc, CustomSectionsList, &m_alloc);
+    }
+    m_customSections->Add(customSection);
+}
+
+uint32
+WebAssemblyModule::GetCustomSectionCount() const
+{
+    return m_customSections ? (uint32)m_customSections->Count() : 0;
+}
+
+Wasm::CustomSection
+WebAssemblyModule::GetCustomSection(uint32 index) const
+{
+    if (index >= GetCustomSectionCount())
+    {
+        throw Wasm::WasmCompilationException(_u("Custom section index out of bounds %u"), index);
```

Comment:
we shouldn't call with out of bounds length, and even if we did this exception kind would be invalid (it isn't a JS exception so it would go uncaught)